docs/core/tracer.md

```diff
@@ -407,6 +407,85 @@ Use **`POWERTOOLS_TRACER_CAPTURE_ERROR=false`** environment variable to instruct
 
     1. You might **return sensitive** information from exceptions, stack traces you might not control
 
+### Reusing Tracer across your code
+
+Tracer keeps a copy of its configuration after the first initialization. This is useful for scenarios where you want to use Tracer in more than one location across your code base.
+
+=== "index.ts"
+
+    ```typescript hl_lines="5"
+    import { Tracer } from '@aws-lambda-powertools/tracer';
+    import middy from '@middy/core';
+    import collectPayment from './payment';
+
+    const tracer = new Tracer(); // (1)
+
+    export const handler = middy(async (event: any, _context: any): Promise<void> => {
+        await collectPayment(event.chargeId);
+    }).use(captureLambdaHandler(tracer));
+    ```
+
+    1.  Because of the way ESM modules work, in order to set up Tracer, you'll need to pass the configurations as [environment variables](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/#environment-variables).
```

Comment:
This line sounds like I cannot pass any parameter to Tracer object.  I can only use environment variable.

That isn't the intention, is it?

Reply:
That's exactly what I'm hinting at. I was going to raise this during the next meeting but essentially what happens is the following:

Suppose we have a file `handler.ts` which is the main entry point for Lambda:
```ts
import { Tracer } from '@aws-lambda-powertools/tracer';
import payment from './payment.ts`

const tracer = new Tracer({ ... some parameters });

export handler = () => {
   payment()
}
```
and then another file `payment.ts`:
```ts
import { Tracer } from '@aws-lambda-powertools/tracer';
import payment from './payment.ts`

const tracer = new Tracer({ ... some OTHER parameters });

const payment = () => {}
```

What will happen in this case is that customers **can pass parameters** to either of the two instances, but while you'd expect the ones in `handler.ts` to be picked up, the instance in `payment.ts` is instead the one that is initialised first and so these settings are picked up and reused.

This happens, to the best of my knowledge, because the module is imported (and so executed) before and even if the order of the imports is changed (i.e. I initialise the "main" tracer before `import payment...`), when using something like `esbuild` the imports are still placed at the top.

With that in mind we have three options:
1. Let customers pass the settings when initialising, but explain this behaviour. This would be in line with the current behaviour that other examples and use cases have but now customers would have to handle the cognitive load of keeping track of which instance is initialised first.
2. Specify that when reusing the tracer across file, the only way to pass configurations is via environment variables. This would mean a different behaviour but since the configurations are in the environment whichever instance is initialised first will always get the correct settings.
3. Push back this feature until we find a better option (if it exists). While evaluating this last option, we should also consider that this same issue will likely affect the other utilities.

Happy to discuss this and definitely open to ideas/suggestions/alternatives.

Reply:
cc @saragerion @flochaz

Reply:
Resolving this conversation in light of the decision detailed [here](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/767#issuecomment-1104995642).